scripts/BadgeManager.js

```diff
@@ -2,20 +2,38 @@
 
 'use strict';
 
-function BadgeManager(addEventListener) {
+function BadgeManager(addEventListener, isEnabled) {
 
 	addEventListener("hourMusic", function(hour) {
-		chrome.browserAction.setBadgeText({ text: formatHour(hour) });
+		if(isEnabled()) {
+			chrome.browserAction.setBadgeText({ text: formatHour(hour) });
+		}
+		playIcon();
 	});
 
 	addEventListener("kkStart", function() {
-		chrome.browserAction.setBadgeText({ text: "KK" });
+		if(isEnabled()) {
+			chrome.browserAction.setBadgeText({ text: "KK" });
+		}
+		playIcon();
 	});
 
 	addEventListener("pause", function() {
 		chrome.browserAction.setBadgeText({ text: "" });
+		pauseIcon();
 	});
+	
+	function playIcon() {
+		chrome.browserAction.setIcon({
+			path : "img/icon_38_leaf-02.png"
```

Comment:
The function names are confusing. I renamed them to be clearer, and made it show KK during his time.